packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -348,6 +348,21 @@ export class EcsRunTask extends sfn.TaskStateBase implements ec2.IConnectable {
     return policyStatements;
   }
 
+  private getTaskDefinitionFamilyArn(): string {
+    let { resource, service, sep, resourceName } = cdk.Stack.of(this).parseArn(this.props.taskDefinition.taskDefinitionArn);
```

Comment:
why the let assignment on all the components of the `ArnComponents`?
you're only inspecting/expecting to modify `resourceName` if necessary.